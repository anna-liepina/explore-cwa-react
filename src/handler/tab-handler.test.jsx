import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent } from '@testing-library/react';
import TabHandler from './tab-handler';

describe('<TabHandler/>', () => {
    const props = {
        tabs: [
            {
                label: 'tab 0',
                c: (props) => <span className='tab0' {...props} />,
            },
            {
                label: 'tab 1',
                c: (props) => <span className='tab1' {...props} />,
                props: {
                    'data-cy': '{{data-cy}}',
                }
            },
        ],
    };

    describe('render', () => {
        it('with default/required props', () => {
            const { asFragment } = render(<TabHandler {...props} />);

            expect(asFragment()).toMatchSnapshot();
        });

        it('with default/required props with preset tabId', () => {
            const { asFragment } = render(<TabHandler {...props} tabId={1} />);

            expect(asFragment()).toMatchSnapshot();
        });
    });

    describe('external callbacks', () => {
        describe('onChange', () => {
            it('should be invoked from a click on a tab [data-cy="tab-$ID"]', () => {
                const spy = jest.fn();
                const { container } = render(<TabHandler {...props} onChange={spy} />);

                fireEvent.click(container.querySelector('[data-cy="tab-1"]'));

                expect(spy).toHaveBeenCalled();
            });
        });
    });
});
