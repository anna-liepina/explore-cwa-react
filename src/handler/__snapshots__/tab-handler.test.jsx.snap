// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabHandler/> render with default/required props 1`] = `
<DocumentFragment>
  <ul
    class="tab-handler"
  >
    <li
      class="tab--selected"
      data-cy="tab-0"
      data-tab-id="0"
    >
      tab 0
    </li>
    <li
      class="tab"
      data-cy="tab-1"
      data-tab-id="1"
    >
      tab 1
    </li>
  </ul>
  <span
    class="tab0"
  />
</DocumentFragment>
`;

exports[`<TabHandler/> render with default/required props with preset tabId 1`] = `
<DocumentFragment>
  <ul
    class="tab-handler"
  >
    <li
      class="tab"
      data-cy="tab-0"
      data-tab-id="0"
    >
      tab 0
    </li>
    <li
      class="tab--selected"
      data-cy="tab-1"
      data-tab-id="1"
    >
      tab 1
    </li>
  </ul>
  <span
    class="tab1"
    data-cy="{{data-cy}}"
  />
</DocumentFragment>
`;
